library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/H262Reader.java

```diff
@@ -130,7 +148,25 @@ public void consume(ParsableByteArray data) {
           hasOutputFormat = true;
         }
       }
+      if (userDataReader != null && userData != null) {
+        int lengthToStartCode = startCodeOffset - offset;
+        int bytesAlreadyPassed = 0;
+        if (lengthToStartCode > 0) {
+          userData.appendToNalUnit(dataArray, offset, startCodeOffset);
+        } else {
+          bytesAlreadyPassed = -lengthToStartCode;
+        }
+
+        if (userData.endNalUnit(bytesAlreadyPassed)) {
```

Comment:
In the H264 case when dealing with SEI, we call `NalUnitUtil.unescapeStream` here. Are you certain it's not necessary to do unescaping here also? The equivalent would be something like:
```
int unescapedLength = NalUnitUtil.unescapeStream(userData.nalData, userData.nalLength);
userDataParsable.reset(sei.nalData, unescapedLength);
userDataParsable.setPosition(4); // NAL prefix and nal_unit() header.
userDataReader.consume(sampleTimeUs, userDataParsable);
```

Reply:
[I have no idea what the answer to this is; I was just comparing the two whilst reviewing]

Reply:
done